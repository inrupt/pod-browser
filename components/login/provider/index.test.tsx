import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

import { login } from "../../../lib/solid-auth-fetcher/dist";

import getProviders from "../../../constants/provider";
import getConfig from "../../../constants/config";

import ProviderLogin, { loginWithProvider } from "./index";

jest.mock("../../../lib/solid-auth-fetcher/dist");

describe("ProviderLogin form", () => {
  test("Renders a webid login form, with button bound to login", () => {
    const tree = shallow(<ProviderLogin />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});

describe("loginWithProvider", () => {
  test("Calls login with clientId if useClientId is true", async () => {
    const oidcIssuer = getProviders()[0];
    const clientId = getConfig().idpClientId;

    await loginWithProvider({
      ...oidcIssuer,
      useClientId: true,
    });

    expect(login).toHaveBeenCalledWith({
      clientId,
      oidcIssuer: oidcIssuer.value,
      redirect: "/loginSuccess",
    });
  });

  test("Calls login without clientId if useClientId is true", async () => {
    const oidcIssuer = getProviders()[0];
    const clientId = getConfig().idpClientId;

    await loginWithProvider({
      ...oidcIssuer,
      useClientId: false,
    });

    expect(login).toHaveBeenCalledWith({
      oidcIssuer: oidcIssuer.value,
      redirect: "/loginSuccess",
    });
  });
});
