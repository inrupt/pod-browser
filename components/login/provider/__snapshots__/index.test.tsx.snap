// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProviderLogin form Renders a webid login form, with button bound to login 1`] = `
<form
  onSubmit={[Function]}
>
  <Styled(MuiBox)
    my={2}
  >
    <WithStyles(ForwardRef(Select))
      onChange={[Function]}
      value="ESS Demo"
    >
      <WithStyles(ForwardRef(MenuItem))
        key="ESS Demo"
        value="ESS Demo"
      >
        ESS Demo
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        key="dev.inrupt.net"
        value="dev.inrupt.net"
      >
        dev.inrupt.net
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        value="other"
      >
        Other
      </WithStyles(ForwardRef(MenuItem))>
    </WithStyles(ForwardRef(Select))>
    <Styled(MuiBox)
      mt={2}
    >
      <WithStyles(ForwardRef(Button))
        color="primary"
        type="submit"
        variant="contained"
      >
        Log In
      </WithStyles(ForwardRef(Button))>
    </Styled(MuiBox)>
  </Styled(MuiBox)>
</form>
`;
